### DISCUSSION TITLE
wire:model -> JS-Error "Uncaught Error: Input must have uuid"

### DISCUSSION THREAD

**Discussion by subIT1** [livewire/livewire#5453] [1. Help]
I have a component which contains an input field with backend live validation.
When typing into the field I get the following error - which counts up:
![image](https://user-images.githubusercontent.com/15226996/208318411-6c2e2df3-d2ef-4165-9813-b7a04a6a3028.png)

Additionaly the blade @error does not remove the Error-Span of the input.

**The visual input-Field with the Error-Text**:
![image](https://user-images.githubusercontent.com/15226996/208318516-edff01a4-4ed9-43b7-9ba5-8203e5149a49.png)

**The input-Field-Component**:
```
<div class="w-full">
    <input id="{{ $id }}"
           name="{{ $id }}"
           @if($model !== null)
               wire:model="{{ $model }}"
           @endif
           class="..."
           value="{{ old($id) }}">
    @if($model !== null)
        @error($model)
        <x-span.error-text>{{ $message }}</x-span.error-text>
        @enderror
    @endif
</div>
```

The field is in a form with `wire:submit.prevent`
and the backend validates via `$this->validate()` on submit.

I also tryed the `wire:key` but nothing changed.

**Comment by joshhanley** [comment]
@subIT1 are you using a third party input library? As that's not an error I've seen before in Livewire's javascript.

**Reply by subIT1** [reply]
@joshhanley No, for the handling I only use the combination of Laravel, Livewire and AlpinJs (+ https://github.com/livewire/turbolinks with https://github.com/hotwired/turbo (for SPA)). 

So what I've found is that it must definitely be the wire:model and that there is no error with a .defer on its "live" binding.
The model handles the change detection via the magic function "updated" and that is where the error seems to come from.

The file "content.js" which is defined in the error message does not exist in the project - so it should be a on the fly generated file.

```
public function updated($propertyName): void
{
        $this->validateOnly($propertyName);
}
```

**Reply by joshhanley** [reply]
@subIT1 can try disabling turbo and seeing if the issue still happens? If it does still happen, any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it? Thanks!

**Reply by subIT1** [reply]
@joshhanley Here is a repo which defines a very minimalistic structure where the error also occours. 
As information: I have the handling in a dialog. Before adding the handling via dialog I got a correct validation without that error. After adding it to the dialog it first seemed to work but when typeing fast the error comes up and stays there. Normally it should not be caused by the dialog structure - cause i got this issue in my main project as well in a non dialog context. 

The view of the issue example Project:
![image](https://user-images.githubusercontent.com/15226996/213874851-dc5bd7d9-0c61-4ee5-bee4-17d8ef045dc4.png)

The issue example project:
[ErrorInput.zip](https://github.com/livewire/livewire/files/10472475/ErrorInput.zip)

Reminder you need a DB cause I used Jetstream in the Project.

**Reply by joshhanley** [reply]
@subIT1 can you publish that to a public GitHub repo? Prefer not to download zip files 🙂 thanks!

**Reply by subIT1** [reply]
@joshhanley understandable ^^ .. [Github-Repo](https://github.com/subIT1/issueLivewire)

**Reply by joshhanley** [reply]
@subIT1 just downloaded your repo and got it running, but I couldn't replicate the issue.

When I typed into the dialog box (even fast), it all seems happy and validation ran as expected on the email field.

 Could you provide me with some steps to follow so I can replicate it? Thanks! 🙂

**Reply by subIT1** [reply]
Hmm that's strange cause I created that issue direcly by opening the dialog and typing in a random text. - The Error at my side occurs one time after x times of testing the input -> reload -> input etc. But when the error occurred once - the error apperas for every furture input (even when reloading the page).

**Reply by joshhanley** [reply]
@subIT1 can you provide me step-by-step instructions, which route to go to, what buttons to click on, what fields to type, what to type, etc. that you are doing so I can follow it exactly? Or create a screen capture of you doing it? I use https://gifcap.dev/

Thanks!

**Reply by subIT1** [reply]
@joshhanley Here is the video. The route is the default and only existing Route "/" of the domain which is the entry point of e.g. "php artisan serve". And as you can see: As far as I typed something in and blur the input field - the console throws the error and counts up every second.

https://user-images.githubusercontent.com/15226996/218860424-ca24fd6e-6986-44cf-a627-cc31d60d2588.mp4

**Reply by joshhanley** [reply]
@subIT1 thanks! This is what happens on my install, all looks ok when I run it.

![Recording 2023-02-15 at 09 27 33](https://user-images.githubusercontent.com/882837/218877587-51261738-94d5-4634-836a-fcbb88b7a3ae.gif)

**Reply by subIT1** [reply]
Very interesting. Any idea how something like this can happen? - cause the same project/code should follow the same result.

**Reply by joshhanley** [reply]
@subIT1 can you test the site in an incognito window or on another computer? Maybe you have browser extensions that are actually causing the error?

**Reply by subIT1** [reply]
@joshhanley The error occoured even on another machine in the incognoto Browser. 
![image](https://user-images.githubusercontent.com/15226996/219458226-9ab6f8bd-4dbe-4589-a93c-85d7f9e60456.png)

**Reply by joshhanley** [reply]
@subIT1 hmm ok maybe I've done something wrong. Can you provide me steps I need to follow, from cloning your repo, installing it, running vite, etc. so I can follow your steps exactly to see if I can replicate it?

**Comment by Dimimo** [comment]
@subIT1 (cc @joshhanley )  do you use NordPass (https://nordpass.com/)? This program comes with NordVPN, a company from Finland and a VPN service I use. NordPass comes with an optional extension (brave://extensions/?id=fooolghllnmhmmndgjiamiiodkpenpbb) - yes, I use Brave, not Chrome

This extension injects a javascript file called content.js, I presume, a helper to inject passwords from the NordPass vault. 

I have the same error and could only find a reference here (I do use LiveWire by the way) This error is NordPass extension related and has nothing to do with LiveWire. If you really want to get rid of it, add `data-np-uid="whatever-text-you-want-here"` to the input element. Just for fun. Of course not for production, it would clutter your rendered HTML with useless baggage. 

Ergo: this is NOT LiveWire related.

**Reply by gijsbeijer** [reply]
For me it was also NordPass firing these errors. Thanks for this!
